xml/src/main/java/org/ehcache/xml/XmlConfiguration.java

```diff
@@ -419,7 +479,19 @@ public URL getURL() {
       throw new IllegalArgumentException("CacheTemplate '" + name + "' declares value type of " + cacheTemplate.valueType());
     }
 
-    CacheConfigurationBuilder<K, V> builder = CacheConfigurationBuilder.newCacheConfigurationBuilder(keyType, valueType);
+    if (resourcePools == null && cacheTemplate.resourcePools().isEmpty()) {
```

Comment:
Strictly speaking, one can have a `ResourcePools` instance that is empty and, to check for that condition, you'd need `resourcePools.getResourceTypeSet().isEmpty()`.